### PR TITLE
go_1_15: 1.15.12 -> 1.15.13, go_1_16: 1.16.4 -> 1.16.5

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Might try to land this at the same time as the `ldflags` PR to save on rebuilds.

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg build go_1_16

File: pkgs/development/compilers/go/binary.nix [hunk 1]
```diff
@@ -29,6 +29,7 @@ stdenv.mkDerivation rec {
   dontStrip = stdenv.hostPlatform.isDarwin;
 
   installPhase = ''
+    runHook preInstall
```
Comment (zowoq) [inline pkgs/development/compilers/go/binary.nix:32]: `nixpkgs-hammering` will probably flag this a some point, adding them in this PR as they cause a full `go_*` rebuild.

**Review by bobrik — APPROVED** [review]

File: pkgs/development/compilers/go/binary.nix (hunk 1 above)
Comment (zowoq) [reply pkgs/development/compilers/go/binary.nix:32]: https://github.com/NixOS/nixpkgs/pull/125745

**Comment by AndersonTorres** [comment]
Can it be backported?

**Pull request merged**